extmod/re1.5/compilecode.c

```diff
@@ -54,6 +54,10 @@ static const char *_compilecode(const char *re, ByteProg *prog, int sizecode)
             prog->len++;
             for (cnt = 0; *re != ']'; re++, cnt++) {
                 if (!*re) return NULL;
+                if (*re == '\\') {
+                    ++re;
+                    if (!*re) return NULL;
```

Comment:
I know the rest of the file uses this style as well, but new code should probably adhere to the hous style i.e. use brackets?

Reply:
Considering this file is from an external module, I would think it best to maintain the existing style.

Reply:
Yeah, not sure, sort of depends on my other question about the relation to upstream, if any.

Anyway: the code can be simplified I think, move the increment before the existing NULL check and only one check is needed:

            for (cnt = 0; *re != ']'; re++, cnt++) {
                if (*re == '\\') {
                    ++re;
                }
                if (!*re) return NULL;

Reply:
Yes, good catch! Thanks

Regarding upstream... it already has diverged.

My intention was to match the existing code style.